app/livechat/imports/server/rest/inquiries.ts

```diff
@@ -1,27 +1,28 @@
 import { Meteor } from 'meteor/meteor';
-import { Match, check } from 'meteor/check';
 
+import { ILivechatInquiryRecord } from '../../../../../definition/IInquiry';
+import { Users, LivechatDepartment, LivechatInquiry } from '../../../../models/server/raw/index';
+import { hasPermission } from '../../../../authorization/client/hasPermission';
 import { API } from '../../../../api/server';
-import { hasPermission } from '../../../../authorization';
-import { Users, LivechatDepartment, LivechatInquiry } from '../../../../models';
 import { findInquiries, findOneInquiryByRoomId } from '../../../server/api/lib/inquiries';
 
 API.v1.addRoute(
 	'livechat/inquiries.list',
 	{ authRequired: true },
 	{
-		get() {
+		async get() {
 			if (!hasPermission(this.userId, 'view-livechat-manager')) {
 				return API.v1.unauthorized();
 			}
 			const { offset, count } = this.getPaginationItems();
 			const { sort } = this.parseJsonQuery();
 			const { department } = this.requestParams();
-			const ourQuery = Object.assign({}, { status: 'queued' });
+			const ourQuery: ILivechatInquiryRecord = Object.assign({}, { status: 'queued', department: 0 });
+
 			if (department) {
-				const departmentFromDB = LivechatDepartment.findOneByIdOrName(department);
-				if (departmentFromDB) {
-					ourQuery.department = departmentFromDB._id;
+				const departmentFromDB: Error = LivechatDepartment.findOneByIdOrName(department);
```

Comment:
🤔 this should return an `ILivechatDepartment` instnce, not an error, let's check why it's returning an error here tomorrow

Reply:
I used Error as it return type because this function just throws an error and nothing else ```findOneByIdOrName(_department: number): Error {
		throw new Error('Method not implemented.');
	}```

:arrow_up: server/raw/LivechatDepartment.ts

Reply:
🤔 I cannot find that, let me see